File: ChipmeoApis/ChipmeoApis.Core/Entities/Discount.cs
namespace ChipmeoApis.Core.Entities;

public partial class Discount
{
    public int Id { get; set; }

    public string Code { get; set; } = null!;

    public string Name { get; set; } = null!;



    public string Type { get; set; } = null!;

    public decimal Value { get; set; }

    public decimal? MaxDiscountAmount { get; set; }

    public decimal? MinOrderAmount { get; set; }

    public int? UsageLimit { get; set; }

    public int? UsedCount { get; set; }

    public bool? IsActive { get; set; }

    public DateTime? StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<Order> Orders { get; set; } = new List<Order>();
}

File: ChipmeoPOS/src/lib/types/discount.ts
export interface Discount {
	id: number;
	code: string;
	name: string;
	description?: string;
	type: 'percent' | 'amount';
	value: number;
	maxDiscountAmount?: number;
	minOrderAmount?: number;
	usageLimit?: number;
	usedCount?: number;
	startDate?: string;
	endDate?: string;
	isActive: boolean;
	createdAt?: string;
	updatedAt?: string;
}

export interface DiscountCreateDto {
	code: string;
	name: string;
	description?: string;
	type: 'percent' | 'amount';
	value: number;
	maxDiscountAmount?: number | null;
	minOrderAmount?: number | null;
	usageLimit?: number | null;
	startDate?: string | null;
	endDate?: string | null;
	isActive: boolean;
}

export interface DiscountUpdateDto {
	code: string;
	name: string;
	description?: string;
	type: 'percent' | 'amount';
	value: number;
	maxDiscountAmount?: number | null;
	minOrderAmount?: number | null;
	usageLimit?: number | null;
	startDate?: string | null;
	endDate?: string | null;
	isActive: boolean;
}
